### PR TITLE
RM-8121 Change type of WebLinkButton.Text from System.String to Remotion.WebString

### PR DESCRIPTION
**Pull request by derlistigelurch**
https://re-motion.atlassian.net/browse/RM-8121

### CODE REVIEW

File: Remotion/Web/Core/UI/Controls/WebLinkButton.cs [hunk 1]
```diff
@@ -53,25 +44,42 @@ protected override void RenderContents (HtmlTextWriter writer)
       if (WcagHelper.Instance.IsWcagDebuggingEnabled() && WcagHelper.Instance.IsWaiConformanceLevelARequired())
         WcagHelper.Instance.HandleError (1, this);
 
+      Text.Write(writer);
+
       if (HasControls())
         base.RenderContents (writer);
       else
-        writer.Write (HotkeyFormatter.FormatText (_textWithHotkey!, false)); // TODO RM-8118: not null assertion
+        writer.Write (AccessKey, false); // TODO RM-8118: not null assertion
```
Comment (iam-mholle) [inline Remotion/Web/Core/UI/Controls/WebLinkButton.cs:52]: I don't think the suggestion can be applied automatically, please fix in a manual commit.

```suggestion
      if (HasControls())
        base.RenderContents (writer);
      else
        Text.Write(writer);
```

File: Remotion/Web/Core/UI/Controls/WebLinkButton.cs [hunk 2]
```diff
@@ -53,25 +44,42 @@ protected override void RenderContents (HtmlTextWriter writer)
...
     }
 
     public new IPage? Page
     {
       get { return PageWrapper.CastOrCreate (base.Page); }
     }
 
-    protected virtual IServiceLocator ServiceLocator
+    [Bindable(true)]
```
Comment (iam-mholle) [inline Remotion/Web/Core/UI/Controls/WebLinkButton.cs:60]:
```suggestion
```

File: Remotion/Web/Core/UI/Controls/WebLinkButton.cs [hunk 3]
```diff
@@ -53,25 +44,42 @@ protected override void RenderContents (HtmlTextWriter writer)
...
+    [DefaultValue("")]
```
Comment (iam-mholle) [inline Remotion/Web/Core/UI/Controls/WebLinkButton.cs:61]: What about `Description`, `Category`?

File: Remotion/Web/Core/UI/Controls/WebLinkButton.cs [hunk 4]
```diff
@@ -53,25 +44,42 @@ protected override void RenderContents (HtmlTextWriter writer)
...
+    public new WebString Text
     {
-      get { return SafeServiceLocator.Current; }
-    }
+      get
+      {
+        var value = (string?) ViewState[c_webLinkButtonTextKey];
+        var type = (WebStringType?) ViewState[c_webLinkButtonTypeKey];
+        type ??= WebStringType.PlainText;
```
Comment (iam-mholle) [inline Remotion/Web/Core/UI/Controls/WebLinkButton.cs:68]:
```suggestion
        var type = (WebStringType?) ViewState[c_webLinkButtonTypeKey] ?? WebStringType.PlainText;
```

**Review by iam-mholle — CHANGES_REQUESTED** [review]

File: Remotion/Web/Core/UI/Controls/WebLinkButton.cs [hunk 5]
```diff
@@ -31,19 +28,13 @@ namespace Remotion.Web.UI.Controls
...
-    private TextWithHotkey? _textWithHotkey;
-
-    protected override void Render (HtmlTextWriter writer)
-    {
-      _textWithHotkey = HotkeyParser.Parse (Text);
-
-      base.Render (writer);
-    }
+    private const string c_webLinkButtonTextKey = nameof (Text);
+    private const string c_webLinkButtonTypeKey = c_webLinkButtonTextKey + nameof (WebStringType);
```
[the remark this answers is not in the document]
Comment (derlistigelurch) [reply Remotion/Web/Core/UI/Controls/WebLinkButton.cs:32]: ok, let's take the pattern you used in #247

**Review by iam-mholle — APPROVED** [review]

**Review by MichaelKetting — APPROVED** [review]

**Pull request merged**